File: two_sum.go
package neetcode

func twoSum(nums []int, target int) []int {
	ind := make(map[int]int)

	for i, n := range nums {
		w := target - n

		if j, ok := ind[w]; ok {
			return []int{i, j}
		} else {
			ind[n] = i
		}
	}

	return nil
}

File: two_sum.rb
# frozen_string_literal: true

def two_sum(nums, target)
  num_indices = {}

  nums.each_with_index do |num, i|
    diff = target - num
    matching_index = num_indices[diff]

    return [i, matching_index] if matching_index

    num_indices[num] = i
  end

  []
end
